test/e2e/self_hosted.go

```diff
@@ -192,6 +192,7 @@ func SelfHostedSpec(ctx context.Context, inputGetter func() SelfHostedSpecInput)
 			InfrastructureProviders:   input.E2EConfig.InfrastructureProviders(),
 			IPAMProviders:             input.E2EConfig.IPAMProviders(),
 			RuntimeExtensionProviders: input.E2EConfig.RuntimeExtensionProviders(),
+			AddonProviders:            input.E2EConfig.AddonProviders(),
```

Comment:
I cannot mark it in this PR but I believe you also need to similarly add addon providers in `clusterctl_upgrade.go` tests.